File: 0208.rb
def is_palindrome(s)
  clean_s = s.downcase.gsub(/[^a-z0-9]/, '')
  clean_s == clean_s.reverse
end

# Example usage:
puts is_palindrome("A man, a plan, a canal: Panama")  # true
puts is_palindrome("race a car")                      # false
puts is_palindrome(" ")                                # true

File: 0208.js
function isPalindrome(s) {
    const cleanS = s.toLowerCase().replace(/[^a-z0-9]/g, '');
    return cleanS === cleanS.split('').reverse().join('');
  }
  
  // Example usage:
  console.log(isPalindrome("A man, a plan, a canal: Panama"));  // true
  console.log(isPalindrome("race a car"));                      // false
  console.log(isPalindrome(" "));                                // true